ruma-common/src/push/predefined_push_rules.rs

```diff
@@ -0,0 +1,251 @@
+            rule_id: ".m.rules.contains_user_name".into(),
+            enabled: true,
+            default: true,
+            pattern: user_id.localpart().into(),
+            actions: vec![
+                Notify,
+                SetTweak(Tweak::Sound("default".into())),
+                SetTweak(Tweak::Highlight(true)),
+            ],
+        }
+    }
+}
+
+/// Default underrides push rules
+impl ConditionalPushRule {
+    /// Matches any incoming VOIP call.
+    pub fn call() -> Self {
+        Self {
+            rule_id: ".m.rules.call".into(),
+            default: true,
+            enabled: true,
+            conditions: vec![EventMatch { key: "type".into(), pattern: "m.call.invite".into() }],
+            actions: vec![
+                Notify,
+                SetTweak(Tweak::Sound("ring".into())),
+                SetTweak(Tweak::Highlight(false)),
+            ],
+        }
+    }
+
+    /// Matches any encrypted event sent in a room with exactly
+    /// two members. Unlike other push rules, this rule cannot be
+    /// matched against the content of the event by nature of it
+    /// being encrypted. This causes the rule to be an "all or
+    /// nothing" match where it either matches all events that are
+    /// encrypted (in 1:1 rooms) or none.
+    pub fn encrypted_room_one_to_one() -> Self {
+        Self {
+            rule_id: ".m.rules.encrypted_room_one_to_one".into(),
+            default: true,
+            enabled: true,
+            conditions: vec![
+                RoomMemberCount { is: RoomMemberCountIs::from(js_int::uint!(2)) },
+                EventMatch { key: "type".into(), pattern: "m.room.encrypted".into() },
+            ],
+            actions: vec![
+                Notify,
+                SetTweak(Tweak::Sound("default".into())),
+                SetTweak(Tweak::Highlight(false)),
+            ],
+        }
+    }
+
+    /// Matches any message sent in a room with exactly two members.
+    pub fn room_one_to_one() -> Self {
+        Self {
+            rule_id: ".m.rules.room_one_to_one".into(),
+            default: true,
+            enabled: true,
+            conditions: vec![
+                RoomMemberCount { is: RoomMemberCountIs::from(js_int::uint!(2)) },
+                EventMatch { key: "type".into(), pattern: "m.room.message".into() },
+            ],
+            actions: vec![
+                Notify,
+                SetTweak(Tweak::Sound("default".into())),
+                SetTweak(Tweak::Highlight(false)),
+            ],
+        }
+    }
+
+    /// Matches all chat messages.
+    pub fn message() -> Self {
+        Self {
+            rule_id: ".m.rules.message".into(),
+            default: true,
+            enabled: true,
+            conditions: vec![EventMatch { key: "type".into(), pattern: "m.room.message".into() }],
+            actions: vec![Notify, SetTweak(Tweak::Highlight(false))],
+        }
+    }
+    /// Matches all encrypted events. Unlike other push rules,
```

Comment:
There should be a newline before this doc comment. Also it seems to be linewrapped way before 100 chars.

Reply:
I wrapped them manually, my editor is not configured to wrap them after 100 chars I think. If it is important I can re-wrap them.

Reply:
No, not that important. There's bound to be inconsistency anyway until we set up rustfmt to do it (just checked actually, maybe this already works with just a config option).